app/lib/frontend/templates/layout.dart

```diff
@@ -103,40 +101,53 @@ String renderLayoutPage(
 
 String _renderSearchBanner({
   @required PageType type,
-  @required String platform,
-  @required String sdk,
   @required String publisherId,
   @required SearchQuery searchQuery,
 }) {
+  final sp = searchQuery != null
+      ? SearchPreference.fromSearchQuery(searchQuery)
```

Comment:
Don't this override my `runtime` preferences, once I search just once for anything flutter related?

Reply:
I think that would be fine. The alternative seems confusing

Reply:
No, this does not override it as a cookie, but the search query (== the configuration of /flutter/packages) takes precedence over the cookie value. Now, I think we agree that this needs to be structured better for longer-term maintenance.